packages/core/src/hooks/useEthers.ts

```diff
@@ -1,31 +1,71 @@
-import { useWeb3React } from '@web3-react/core'
-import { Web3Provider } from '@ethersproject/providers'
-import { ChainId } from '../constants'
 import { useCallback } from 'react'
-import { useConfig } from '../providers/config/context'
-import { InjectedConnector } from '@web3-react/injected-connector'
-
-type ActivateBrowserWallet = (onError?: (error: Error) => void, throwErrors?: boolean) => void
+import { JsonRpcProvider } from '@ethersproject/providers'
+import { ChainId } from '../constants'
+import { useInjectedProvider, useNetwork } from '../providers'
 
-export type Web3Ethers = ReturnType<typeof useWeb3React> & {
-  library?: Web3Provider
+export type Web3Ethers = {
+  activate: (provider: JsonRpcProvider) => Promise<void>
+  setError: (error: Error) => void
+  deactivate: () => void
+  connector: undefined
   chainId?: ChainId
-  activateBrowserWallet: ActivateBrowserWallet
+  account?: null | string
+  error?: Error
+  library?: JsonRpcProvider
+  active: boolean
+  activateBrowserWallet: () => void
+}
+
+async function tryToGetAccount(provider: JsonRpcProvider) {
+  try {
+    return await provider.getSigner().getAddress()
+  } catch {
+    return undefined
+  }
 }
 
 export function useEthers(): Web3Ethers {
-  const result = useWeb3React<Web3Provider>()
-  const { networks } = useConfig()
-  const activateBrowserWallet = useCallback<ActivateBrowserWallet>(
-    async (onError, throwErrors) => {
-      const injected = new InjectedConnector({ supportedChainIds: networks?.map((network) => network.chainId) })
-      if (onError instanceof Function) {
-        await result.activate(injected, onError, throwErrors)
-      } else {
-        await result.activate(injected, undefined, throwErrors)
-      }
+  const {
+    network: { provider, chainId, accounts, errors },
+    update,
+  } = useNetwork()
+  const { injectedProvider, connect } = useInjectedProvider()
+
+  const result = {
+    connector: undefined,
+    library: provider,
+    chainId,
+    account: accounts[0],
```

Comment:
btw. would be good to add `accounts`

Reply:
Yes it will, but later. For now I was aiming on not introducing breaking changes